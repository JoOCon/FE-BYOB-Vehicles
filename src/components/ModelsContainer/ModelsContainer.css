.makes-cont {
  display: flex;
  margin: 1rem;
  overflow: scroll;
}