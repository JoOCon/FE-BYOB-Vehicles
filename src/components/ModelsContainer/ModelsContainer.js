import React from 'react';
import Card from '../Card/Card';
import './ModelsContainer.css';

const ModelsContainer = (props) => {
  const { deleteModel, models } = props;
  const makesList = models.map(model => (
    <Card
      {...model}
      deleteModel={deleteModel}
      key={model.id}
    />
  ));
  return (
    <section className="makes-cont">
      {makesList}
    </section>
  );
};

export default ModelsContainer;
