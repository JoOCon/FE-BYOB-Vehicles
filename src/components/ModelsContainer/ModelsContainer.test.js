import React from 'react';
import { shallow } from 'enzyme';
import ModelsContainer from './ModelsContainer';

describe('ModelsContainer', () => {
  it('should match snapshot with models', () => {
    const mockModels = [{ name: 'garry' }, { name: 'billy' }];
    const wrapper = shallow(
      <ModelsContainer
        models={mockModels}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
