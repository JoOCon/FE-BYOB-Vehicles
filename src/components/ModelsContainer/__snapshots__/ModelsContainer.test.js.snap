// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModelsContainer should match snapshot with models 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ModelsContainer
    models={
      Array [
        Object {
          "name": "garry",
        },
        Object {
          "name": "billy",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          name="garry"
        />,
        <Card
          name="billy"
        />,
      ],
      "className": "makes-cont",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "name": "garry",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "name": "billy",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            name="garry"
          />,
          <Card
            name="billy"
          />,
        ],
        "className": "makes-cont",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "name": "garry",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "name": "billy",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
